packages/components/src/custom-select-control-v2/legacy-component/test/index.tsx

```diff
@@ -64,7 +64,7 @@ const ControlledCustomSelectControl = ( {
 	onChange: onChangeProp,
 	...restProps
 }: React.ComponentProps< typeof UncontrolledCustomSelectControl > ) => {
-	const [ value, setValue ] = useState( options[ 0 ] );
+	const [ value, setValue ] = useState( restProps.value ?? options[ 0 ] );
```

Comment:
This change allows the `Controlled` version of the component to use the `value` prop as the initial value (the same change was applied to the v1 component)